### PR TITLE
hdfi.h is no longer included in hdf.h

### PR DESCRIPTION
**Pull request by derobins**
In the past, hdfi.h (an internal header file) was included in the
  public hdf.h header. This is no longer the case. Anything that
  should be public has been moved from hdfi.h to hdf.h, including:

    * The C99-like types (e.g., float32)
    * The archaic VOID/VOIDP macros * The intf Fortran integer type * The HD library macros (e.g., HDmalloc)

  All of these things are considered obsolete and we only maintain
  them in the public header so that old code does not have to be
  modified. It is unwise to use them in new code.

### CODE REVIEW

File: hdf/src/H4api_adpt.h [hunk 1]
```diff
@@ -14,8 +14,6 @@
 #ifndef H4_API_ADPT_H
 #define H4_API_ADPT_H
 
-#include "h4config.h"
-
```
Comment (derobins) [inline hdf/src/H4api_adpt.h:18]: Moved to hdf.h

**Comment by derobins** [comment]
Wherever possible, internal library and test source/headers include hdfi.h. Tools and utility code only include hdf.h, wherever possible.

**Comment by derobins** [comment]
Yay

**Review by byrnHDF — APPROVED** [review]

**Review by bmribler — APPROVED** [review]

File: hdf/src/hdf.h [hunk 2]
```diff
@@ -166,16 +173,20 @@ typedef intn (*hdf_termfunc_t)(void); /* termination function typedef */
...
+ */
 
 #include "hbitio.h"
 #include "hcomp.h"
 #include "herr.h"
 #include "hproto.h"
-#include "hdatainfo.h" /* Add the data info header */
-#include "vg.h"        /* Add the Vgroup/Vdata header so the users don't have to */
-#include "mfgr.h"      /* Add the GR header so the users don't have to */
+#include "hdatainfo.h" /* data info header */
```
[the remark this answers is not in the document]
Comment (derobins) [reply hdf/src/hdf.h:187]: We can remove it in a future PR, when we remove the file.

**Pull request merged**